src/brsTypes/components/RoSGNode.ts

```diff
@@ -151,6 +154,18 @@ export class Field {
         // Once a field is set, it is no longer hidden.
         this.hidden = false;
 
+        if (isBrsNumber(value)) {
```

Comment:
💭 Would it be any cleaner to use `getValueKindFromFieldType` here?
```brightscript
if (isBrsNumber(value) && value.kind !== getValueKindFromFieldType(this.type)) {
  switch (value.kind) {
    case ValueKind.Float:
      value = new Float(value.getValue());
      break;
    case ValueKind.Int32:
...
```

Reply:
The switch adds a lot of break boilerplate IMO, but your suggestion of moving the field check out definitely makes things cleaner. Will change.